ssf/samples.go

```diff
@@ -47,13 +69,46 @@ func TimeUnit(resolution time.Duration) SampleOption {
 	}
 }
 
+// Prefix prepends the given prefix to a sample. Due to the
+// by-order-of-arguments nature of applying these arguments, if more
+// than one Prefix argument is given, the leftmost Prefix argument
+// will be the rightmost name component on the final name.
+func Prefix(prefix string) SampleOption {
+	return func(s *SSFSample) {
+		s.Name = prefix + s.Name
+	}
+}
+
 func create(base *SSFSample, opts []SampleOption) *SSFSample {
+	for _, opt := range DefaultSampleOptions {
+		opt(base)
+	}
 	for _, opt := range opts {
 		opt(base)
 	}
 	return base
 }
 
+// Sampled takes a rate and a set of measurements, and returns a new
+// set of measurements as if sampling had been performed: Each
+// original measurement gets rejected/included in the result based on
+// a random roll of the RNG according to the rate, and each included
+// measurement has its SampleRate field adjusted to be its original
+// SampleRate * rate.
+func Sampled(rate float32, samples ...*SSFSample) []*SSFSample {
```

Comment:
Should this be `Sampled` or `Sample`? "Sample" is the action that it's taking, so that'd make more sense, no?

Reply:
True. I think we're already kinda in a hole here w.r.t. the naming of `SSFSample`; calling another thing "sample" would make it all even more confusing. Maybe we can make it more explicit? `DropSampled` maybe?

Reply:
`RandomlySample`? Endless possibilities, but I don't know if I like any of these /:

Reply:
My main issue with the name `Sampled` is that it's not an imperative verb (which most function names are). The only times I ever really see a past participle in a function name, I expect it to return a boolean (ie, semantically equivalent to `func IsSampled?() bool`).


I'd prefer `RandomlySample` even if it's non-ideal, just because it fits grammatically. Open to alternatives though.

Reply:
RandomSample it is! If somebody points at this, I'll say a statistician signed off on it (-;

Reply:
> 

RandomlySample it is! If somebody points at this, I'll say a statistician signed off on it (-;

Reply:
> RandomlySample it is! If somebody points at this, I'll say a statistician signed off on it (-;


I'll bite my tongue and refrain from insisting that we name it `PseudorandomlySample`! ;)